Upgrade to Scala 2.11 and removal of old API

stephanh opened the pull request:
* Upgrade to Scala 2.11 and compile error and warning fixes
* Remove old Jobs/Flow API since the tests broke as part of the upgrade. This closes #39.
* Added ability to easily test Scalding typed pipes and grouped.

shmookey commented:
Looks good to me.

vineethvarghese commented on README.md:
```diff
@@ -77,97 +83,74 @@ Thermometer expectations allow you to fall back to specs2, this may be because
 
-dependent pipelines
--------------------
+nntesting scalding pipes directly
```
what is nntesting?

stephanh replied:
A typo :smile:

vineethvarghese commented on README.md:
```diff
@@ -189,9 +172,6 @@ ongoing work
- - A ThermometerSink that would allow in memory fact checking
- - Support for running full jobs in the same fact framework
- - Support for re-running tests with different scalding modes
+  - Support for re-running tests with different scalding modes
```
Are we working on these items?

stephanh replied:
No they were in the README from before.

stephanh replied:
Fixed.

vineethvarghese commented on core/src/main/scala/au/com/cba/omnia/thermometer/core/ThermometerSpec.scala:
```diff
@@ -29,48 +29,41 @@ import org.apache.hadoop.fs.{FileSystem, Path}
-    new VerifiableJob(j)
+  /** Ensures that each of the tests is run sequentially and isolated. */
+  override def map(struct: SpecStructure) =
+    sequential ^ isolated ^ struct
```
We could discuss this in person and I believe you have mentioned this before (just that I forgot) but what is limiting us from executing sequentially? From what I understand each Thermometer spec get a hadoop env via the `HadoopSupport` trait. Why not extend this to each test and get parallel tests? Unless there is some fundamental flaw in the Hadoop library like sticking all the env into some static variable, from the thermometer code I don't see (yet) why this can't be done.

stephanh replied:
The Hadoop filesystem is cached and it caches the configuration accross runs.

vineethvarghese replied:
Hmm...it would be interesting but if we "somehow" manage to run each test with its own classloader, we might be able to work around the hadoop problem. But that is a separate problem.

vineethvarghese commented on core/src/main/scala/au/com/cba/omnia/thermometer/core/ThermometerSpec.scala:
```diff
@@ -29,48 +29,41 @@ import org.apache.hadoop.fs.{FileSystem, Path}
+  /** Sets the JVM user.dir to a temporary test location. */
+  def before = {
+    new File(dir, "user").mkdirs()
+    System.setProperty("user.dir", s"${dir}/user")
```
Why did we have to do this with the upgrade?

stephanh replied:
It used to be done as part of the HadoopSupport trait. However, with SPECS3.5 or Scala 2.11 we need to reset the `user.dir` after the test so that is why I moved the code here and arranged it in this way.

vineethvarghese commented:
Nice additions. I have a few question though. About the readme, do we want to mention that we have dropped support for scalding jobs?

stephanh commented:
I don't think the readme should mention that we dropped that support. IMO the readme is there to help you use the project in its current form. Issues, PRs and commits are there to record the evolution of the project.

I changed the major version number to capture the impact of this change.

vineethvarghese commented:
:+1:

stephanh commented:
I also added ThermometerHiveSpec to work around a the fact that both ThermometerSpec and HiveSupport now have a before method.

vineethvarghese commented:
@stephanh Do we want a separate HiveSupport now?

stephanh commented:
@vineethvarghese I think so. It preserves backward compatibility a bit more and you can use it independently of `ThermometerSpec`.

vineethvarghese commented:
@stephanh Fair enough. All that is left then is the build failure otherwise :+1:
